grovedb/src/replication/state_sync_session.rs

```diff
@@ -204,6 +221,14 @@ impl<'db> MultiStateSyncSession<'db> {
         unsafe { Pin::into_inner_unchecked(self) }.transaction
     }
 
+    pub unsafe fn set_new_transaction(
```

Comment:
`pub unsafe` is too much, and no safety docs even

Reply:
@fominok reduced the scope of unsafe.
Is there a better way to set a new transaction without consuming the session?